src/main/java/io/jenkins/plugins/bitbucketpushandpullrequest/client/BitBucketPPRCloudClient.java

```diff
@@ -0,0 +1,112 @@
+/*******************************************************************************
+ * The MIT License
+ * 
+ * Copyright (C) 2020, CloudBees, Inc.
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ ******************************************************************************/
+
+package io.jenkins.plugins.bitbucketpushandpullrequest.client;
+
+import java.nio.charset.StandardCharsets;
+import java.util.logging.Logger;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import com.cloudbees.plugins.credentials.domains.URIRequirementBuilder;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.http.HttpHeaders;
+import org.apache.http.HttpResponse;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import hudson.model.Run;
+import hudson.plugins.git.UserRemoteConfig;
+import io.jenkins.plugins.bitbucketpushandpullrequest.BitBucketPPRTrigger;
+
+
+public class BitBucketPPRCloudClient implements BitBucketPPRClient {
+  private static final Logger LOGGER = Logger.getLogger(BitBucketPPRTrigger.class.getName());
+  private final UserRemoteConfig config;
+  private Run<?, ?> run;
+
+  public BitBucketPPRCloudClient(final UserRemoteConfig config, Run<?, ?> run) {
+    this.run = run;
+    this.config = config;
+  }
+
+  public void sendWithUsernamePasswordCredentials(final String url, String payload) {
+    LOGGER.info("Set BB Credentials");
+    final org.apache.http.client.CredentialsProvider provider = new BasicCredentialsProvider();
```

Comment:
once there are more ways to authenticate (e.g. token based?) it would make probably sense to extract this into a dedicated Authentication class or factory...